packages/common/src/services/Jupiter.ts

```diff
@@ -21,3 +24,38 @@ export const parseJupiterInstruction = (instruction: Instruction) => {
 }
 
 export const jupiterInstance = createJupiterApiClient()
+
+export const quoteWithAnalytics = async ({
```

Comment:
nit: Couldn't we just as easily do the analytics in the saga before/after calling this function? Would mean less dependency injection here.

Reply:
yeah that's how I had it before, but it honestly was getting so verbose and felt nice to have a helper function that could do it all without `yield* call()`s - I also think this will be useful for other flows that do jupiter swaps, I just haven't moved them to this fn in this PR